cuml/forest_inference_demo.ipynb

```diff
@@ -0,0 +1,255 @@
+    "import pytest\n",
+    "import os\n",
+    "\n",
+    "from cuml import ForestInference\n",
+    "from cuml.test.utils import array_equal\n",
+    "from cuml.utils.import_utils import has_xgboost\n",
+    "\n",
+    "from sklearn.datasets import make_classification\n",
+    "from sklearn.metrics import accuracy_score\n",
+    "from sklearn.model_selection import train_test_split\n",
+    "\n",
+    "if has_xgboost():\n",
+    "    import xgboost as xgb\n",
+    "else:\n",
+    "    raise(\"Please install xgboost using the conda package,\"\n",
+    "          \" Use conda install -c conda-forge xgboost \"\n",
+    "          \"command to install xgboost\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### function that trains the xgboost model and performs prediction on it as well\n",
+    "\n",
+    "#### for additional information on the xgboost library please refer to the documentation on : \n",
+    "#### https://xgboost.readthedocs.io/en/latest/parameter.html"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def train_and_predict_xgboost_model(X_train, y_train,\n",
+    "                                    X_validation,\n",
+    "                                    y_validation,\n",
+    "                                    num_rounds, model_path):\n",
+    "    # set the xgboost model parameters\n",
+    "    params = {'silent': 1, 'eval_metric':'error',\n",
+    "              'objective':'binary:logistic',\n",
+    "              'max_depth': 25}\n",
+    "    dtrain = xgb.DMatrix(X_train, label=y_train)\n",
+    "    # train the xgboost model\n",
+    "    bst = xgb.train(params, dtrain, num_rounds)\n",
+    "\n",
+    "    # save the trained xgboost model\n",
+    "    bst.save_model(model_path)\n",
+    "\n",
+    "    # predict the xgboost model\n",
+    "    dvalidation = xgb.DMatrix(X_validation, label=y_validation)\n",
+    "    xgb_preds = bst.predict(dvalidation)\n",
+    "\n",
+    "    # convert the predicted values from xgboost into class labels\n",
+    "    xgb_preds = np.around(xgb_preds)\n",
+    "\n",
+    "    return xgb_preds, model_path"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### setting the required parameters"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "n_rows = 10000\n",
+    "n_columns = 100\n",
+    "n_categories = 2\n",
+    "random_state = np.random.RandomState(43210)\n",
+    "\n",
+    "# enter path to the directory where the trained model will be saved\n",
+    "model_path = 'xgb.model'\n",
+    "# num of iterations for which the model is trained\n",
+    "num_rounds = 15"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### create the dataset and call the function to train and predict the xgboost model"
```

Comment:
2nd-level header (`##`). I've been using `## Generate Data` for a simple header and then putting more details in regular sized text underneath, like this:

```
## Generate Data

Generate the dataset on the host
```

I would also end this cell on the `train_test_split` and put the train/predict call in its own cell, with a heading:

```
## Train and Predict Model

Invoke the function to train the model and get predictions so that we can validate them. 
```